src/auth/user_access_manager.cc

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2021 4Paradigm
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "user_access_manager.h"
+
+#include <unordered_map>
+#include <utility>
+
+#include "auth_utils.h"
+#include "nameserver/system_table.h"
+
+namespace openmldb::auth {
+UserAccessManager::UserAccessManager(IteratorFactory iterator_factory,
+                                     std::shared_ptr<nameserver::TableInfo> user_table_info)
+    : user_table_iterator_factory_(std::move(iterator_factory)), user_table_info_(user_table_info) {
+    StartSyncTask();
+}
+
+UserAccessManager::~UserAccessManager() { StopSyncTask(); }
+
+void UserAccessManager::StartSyncTask() {
+    sync_task_running_ = true;
+    sync_task_thread_ = std::thread([this] {
```

Comment:
query database every 0.1 seconds ?

Reply:
Yeah this is just temporary solution to get latest user credentials. In the future we will use a distributed transaction to update user table cache on all nodes.